pkg/bpf/bpf_linux.go

```diff
@@ -113,7 +114,20 @@ func UpdateElementFromPointers(fd int, structPtr unsafe.Pointer, sizeOfStruct ui
 	}
 
 	if ret != 0 || err != 0 {
-		return fmt.Errorf("Unable to update element for map with file descriptor %d: %s", fd, err)
+		switch err {
+		case unix.E2BIG:
+			if strings.Contains(mapName, "lb") {
```

Comment:
The `pkg/bpf` package should not know anything about packages which use it. Please move this particular error handling to `pkg/maps/lbmap`.

Reply:
@brb  thanks for your review.
>
> The pkg/bpf package should not know anything about packages which use it
>
Could you say more please?
 
>
> Also, matching on map name is very brittle
>
I am aware and agree with you. Please, see my previous comment on the discussion.

Reply:
>> The pkg/bpf package should not know anything about packages which use it

> Could you say more please?

You are introducing a logical circular dependency. Why is it not possible to return this custom error from `pkg/maps/lbmap`?

Reply:
> You are introducing a logical circular dependency.
Why ? Is it because 1) I am getting the mapName as a parameter or because 2) I am matching the mapName is pkg/bpf package?

> Why is it not possible to return this custom error from `pkg/maps/lbmap`?
It is possible. I am trying first to understand why we should proceed like that.

Reply:
`pkg/maps/lbmap` depends on `pkg/bpf` which is a generic library. It should not know any specific details of its consumers. Your proposed change adds a custom logic of `pkg/maps/lbmap` into `pkg/bpf`. So we have https://en.wikipedia.org/wiki/Circular_dependency.

Next, the logic of `pkg/maps/lbmap` is getting widespread across the codebase instead of being self-isolated in its package. This has a side-effect that it becomes more difficult to reason about `pkg/maps/lbmap`.

Reply:
Oki, I get your point. However, it looks impossible to print another message at the place of `argument list too long` without getting any information from pkg/maps/lbmap in pkg/bpf.

Reply:
Couldn't you change the previous return statement to `return fmt.Errorf("Unable to update element for map with file descriptor %d: %w", fd, err)` (diff - `s/%s/%w/`), and then extract the error from `pkg/maps/lbmap/` with `pkg/errors`?

Reply:
Oki. Let me try it.

Reply:
> Couldn't you change the previous return statement to `return fmt.Errorf("Unable to update element for map with file descriptor %d: %w", fd, err)` (diff - `s/%s/%w/`), and then extract the error from `pkg/maps/lbmap` with `pkg/errors`?

If we proceed like that, it will work for `lbmap`. But for the other maps, the message should be the generic `argument list too long`. Unless, we decide to handle the error in the same for all other maps as we do in `pkg/maps/lbmap`.

Reply:
If there is a need, we could handle the same in other map packages as in pkg/maps/lbmap`.